pkg/controllers/queue/queue_controller_test.go

```diff
@@ -242,55 +244,125 @@ func TestSyncQueue(t *testing.T) {
+					Status: schedulingv1beta1.PodGroupStatus{
+						Phase: schedulingv1beta1.PodGroupPending,
+					},
+				},
+			},
+			queue: &schedulingv1beta1.Queue{
 				ObjectMeta: metav1.ObjectMeta{
-					Name:      "pg1",
-					Namespace: namespace,
+					Name: "c1",
 				},
-				Spec: schedulingv1beta1.PodGroupSpec{
-					Queue: "c1",
+				Spec: schedulingv1beta1.QueueSpec{
+					Weight: 1,
 				},
-				Status: schedulingv1beta1.PodGroupStatus{
-					Phase: schedulingv1beta1.PodGroupPending,
+			},
+			ExpectStatus: schedulingv1beta1.QueueStatus{
+				Pending:     1,
+				Reservation: schedulingv1beta1.Reservation{},
+				Allocated:   v1.ResourceList{},
+			},
+		},
+		{
+			Name: "syncQueueHandlingNotFoundPg",
+			podGroups: []*schedulingv1beta1.PodGroup{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "pg1",
+						Namespace: namespace,
+					},
+					Spec: schedulingv1beta1.PodGroupSpec{
+						Queue: "c2",
+					},
+					Status: schedulingv1beta1.PodGroupStatus{
+						Phase: schedulingv1beta1.PodGroupPending,
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "pg2",
+						Namespace: namespace,
+					},
+					Spec: schedulingv1beta1.PodGroupSpec{
+						Queue: "c2",
+					},
+					Status: schedulingv1beta1.PodGroupStatus{
+						Phase: schedulingv1beta1.PodGroupPending,
+					},
+				},
+			},
+			delayDeletedPodGroups: []*schedulingv1beta1.PodGroup{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "pg2",
+						Namespace: namespace,
+					},
+					Spec: schedulingv1beta1.PodGroupSpec{
+						Queue: "c2",
+					},
 				},
 			},
 			queue: &schedulingv1beta1.Queue{
 				ObjectMeta: metav1.ObjectMeta{
-					Name: "c1",
+					Name: "c2",
 				},
 				Spec: schedulingv1beta1.QueueSpec{
 					Weight: 1,
 				},
 			},
-			ExpectValue: 1,
+			ExpectStatus: schedulingv1beta1.QueueStatus{
+				State:       schedulingv1beta1.QueueStateOpen,
+				Pending:     1,
+				Reservation: schedulingv1beta1.Reservation{},
+				Allocated:   v1.ResourceList{},
+			},
 		},
 	}
 
 	for i, testcase := range testCases {
 		c := newFakeController()
 
-		key, _ := cache.MetaNamespaceKeyFunc(testcase.podGroup)
-		c.podGroups[testcase.podGroup.Spec.Queue] = make(map[string]struct{})
-		c.podGroups[testcase.podGroup.Spec.Queue][key] = struct{}{}
+		for j := range testcase.podGroups {
+			key, _ := cache.MetaNamespaceKeyFunc(testcase.podGroups[j])
+			if _, ok := c.podGroups[testcase.podGroups[j].Spec.Queue]; !ok {
+				c.podGroups[testcase.podGroups[j].Spec.Queue] = make(map[string]struct{})
+			}
+			c.podGroups[testcase.podGroups[j].Spec.Queue][key] = struct{}{}
+
+			c.pgInformer.Informer().GetIndexer().Add(testcase.podGroups[j])
+		}
 
-		c.pgInformer.Informer().GetIndexer().Add(testcase.podGroup)
 		c.queueInformer.Informer().GetIndexer().Add(testcase.queue)
 		c.vcClient.SchedulingV1beta1().Queues().Create(context.TODO(), testcase.queue, metav1.CreateOptions{})
 
+		for j := range testcase.delayDeletedPodGroups {
```

Comment:
Repeat add and delete pg in informer is not graceful, you can add two fieleds in testCase like pginInformer and pgInCache to distinguish them: )